src/events.jl

```diff
@@ -44,35 +44,52 @@ end
 ```
 """
 function event_loop(callback::Function, plotobj::DisplazWindow, event_list...)
+  
+    command = hook_command(plotobj.name, event_list...)
+
+    open(command, "r") do stdout
+        handle_events(callback, stdout)
+    end
+
+end
+
+function hook_command(servername, event_list...)
+
     hookopts = [["-hook", _eventspec_str(e), _argspec_str(p)]
-                for (e,p) in event_list]
-    stdout,stdin,proc = readandwrite(`$_displaz_cmd -server $(plotobj.name) $(vcat(hookopts...))`)
-    while true
+    for (e,p) in event_list]
+
+    return `$_displaz_cmd -server $(servername) $(vcat(hookopts...))`
+end
+
+function handle_events(callback, stdout)
+
+    while !eof(stdout)
         rawline = readline(stdout)
+        @show rawline
```

Comment:
```suggestion
```

Or leave it in as `@debug "Raw displaz event line" rawline`

Reply:
Oops - nah, remove is better, thanks!